disnake/ui/action_row.py

```diff
@@ -0,0 +1,251 @@
+
+        Parameters
+        -----------
+        item: :class:`disnake.ui.Item`
+            The item to append to the action row.
+
+        Raises
+        -------
+        ValueError
+            The width of the action row exceeds 5.
+        """
+        if self._total_width + item.width > 5:
+            raise ValueError("Too many items in this row, can not appen a new one.")
+
+        self._total_width += item.width
+        self._underlying.children.append(item._underlying)  # type: ignore
+
+    def add_button(
+        self,
+        *,
+        style: ButtonStyle = ButtonStyle.secondary,
+        label: Optional[str] = None,
+        disabled: bool = False,
+        custom_id: Optional[str] = None,
+        url: Optional[str] = None,
+        emoji: Optional[Union[str, Emoji, PartialEmoji]] = None,
+    ):
+        """Adds a button to the action row.
+
+        To append a pre-existing :class:`disnake.ui.Button` use the
+        :meth:`append_item` method instead.
+
+        Parameters
+        -----------
+        style: :class:`disnake.ButtonStyle`
+            The style of the button.
+        custom_id: Optional[:class:`str`]
+            The ID of the button that gets received during an interaction.
+            If this button is for a URL, it does not have a custom ID.
+        url: Optional[:class:`str`]
+            The URL this button sends you to.
+        disabled: :class:`bool`
+            Whether the button is disabled or not.
+        label: Optional[:class:`str`]
+            The label of the button, if any.
+        emoji: Optional[Union[:class:`.PartialEmoji`, :class:`.Emoji`, :class:`str`]]
+            The emoji of the button, if available.
+
+        Raises
+        -------
+        ValueError
+            The width of the action row exceeds 5.
+        """
+        if self._total_width >= 5:
+            raise ValueError("Too many items in this row, can not add a button.")
+
+        self._total_width += 1
+        self._underlying.children.append(
+            ButtonComponent._raw_construct(
+                type=ComponentType.button,
+                custom_id=custom_id,
+                url=url,
+                disabled=disabled,
+                label=label,
+                style=style,
+                emoji=emoji,
+            )
+        )
+
+    def add_select(
+        self,
+        *,
+        custom_id: str = MISSING,
+        placeholder: Optional[str] = None,
+        min_values: int = 1,
+        max_values: int = 1,
+        options: List[SelectOption] = MISSING,
+        disabled: bool = False,
+    ):
+        """Adds a select menu to the action row.
+
+        To append a pre-existing :class:`disnake.ui.Select` use the
+        :meth:`append_item` method instead.
+
+        Parameters
+        -----------
+        custom_id: :class:`str`
+            The ID of the select menu that gets received during an interaction.
+            If not given then one is generated for you.
```

Comment:
Raw components don't generate random `custom_id`s (also see above)

Reply:
Oops I copied the docs from button.py but used raw_construct instead